gencode.c

```diff
@@ -9071,13 +9070,16 @@ gen_acode(compiler_state_t *cstate, const char *s, struct qual q)
 	case DLT_ARCNET_LINUX:
 		if ((q.addr == Q_HOST || q.addr == Q_DEFAULT) &&
 		    q.proto == Q_LINK) {
-			cstate->e = pcap_ether_aton(s);
-			if (cstate->e == NULL)
-				bpf_error(cstate, "malloc");
-			b = gen_ahostop(cstate, cstate->e, (int)q.dir);
-			free(cstate->e);
-			cstate->e = NULL;
-			return (b);
+			bpf_u_int32 addr;
+			/*
+			 * The lexer currently defines the address format in a
+			 * way that makes this error condition never true.
+			 * Let's check it anyway in case this part of the lexer
+			 * changes in future.
+			 */
+			if (! pcapint_atoan(s, &addr))
+			    bpf_error(cstate, "invalid ARCnet address '%s'", s);
+			return gen_ahostop(cstate, addr, (int)q.dir);
```

Comment:
This gets

```
C:\projects\libpcap\gencode.c(9082): warning C4242: 'function': conversion from 'bpf_u_int32' to 'const uint8_t', possible loss of data [C:\projects\libpcap\build\pcap.vcxproj]
```

from Visual Studio, which is the *only* C compiler I've ever seen warn about converting a wider integer value to a narrower integer value (except for Apple's GCC/Clang, which do so only for 64-bit -> 32-bit, as part of their 64-bit-ification of macOS).

You probably need to do something such as

```
			if (! pcapint_atoan(s, &addr) || addr > UINT8_MAX)
			    bpf_error(cstate, "invalid ARCnet address '%s'", s);
```

to make sure that it's a value that fits in a 1-byte address field, and then cast `addr` to `uint8_t` to tell the compiler "yes, I've checked, this is OK".

Reply:
Or perhaps `pcapint_atoan()` could take a pointer to an `uint8_t` instead of trying to look consistent with the other functions...